samples: drivers: watchdog: Run sample on nrf54h20

nordic-segl opened the pull request:
Add overlay that enables selected WDT instance on
nrf54h20 Application core.

nordic-segl committed:
samples: drivers: watchdog: Run sample on nrf54h20

Add overlay that enables selected WDT instance on
nrf54h20 Application core.

Signed-off-by: Sebastian Głąb <sebastian.glab@nordicsemi.no>

nordic-segl commented:
FYI: @nika-nordic @anangl @kl-cruz @nordic-piks

nordic-segl commented on samples/drivers/watchdog/boards/nrf54h20dk_nrf54h20_cpuapp.overlay:
```diff
@@ -0,0 +1,14 @@
+
+/ {
+	aliases {
+		watchdog0 = &wdt010;
```
nrf54l15 has this alias in the board definition, see:
https://github.com/zephyrproject-rtos/zephyr/blob/main/boards/nordic/nrf54l15pdk/nrf54l15pdk_nrf54l15-common.dtsi#L63
@anangl: Shall I do the same for nrf54h20 cpuapp? What about cpurad (wdt010 instances are local and multiple cores can use wdt010 as it maps to a different base addresses)?

nordic-segl replied:
Such alias is also crucial to run this sample
https://github.com/zephyrproject-rtos/zephyr/blob/main/samples/subsys/task_wdt/src/main.c#L24

anangl replied:
Yes, please add it in the board definition.

> What about cpurad (wdt010 instances are local and multiple cores can use wdt010 as it maps to a different base addresses)?

I'm not sure if I understand well what you are asking about, but I think it is okay to add same alias for cpurad.

nordic-segl replied:
I moved watchdog0 alias to the board definition for Application core.

I decided to NOT define such alias for Radio core because:
- there are no other aliases defined on that core;
- I don't run samples/drivers/watchdog on Radio core; I don't know if such alias is needed by other samples/tests or just by this one.
- I don't want to pollute DTS with unused aliases. It can always be added later.

nordic-segl committed:
boards: nordic: nrf54h20dk: Define alias watchdog0

Define alias watchdog0 for nrf54h20 Application core.

Signed-off-by: Sebastian Głąb <sebastian.glab@nordicsemi.no>

nordic-segl committed:
boards: nordic: nrf54h20dk: Disable watchdog tests on Radio core

It is enough to test watchdog driver on the application core.
Otherwise, at least three samples/tests fail due to missing
overlay file for nrf54h20dk/nrf54h20/cpurad.

Signed-off-by: Sebastian Głąb <sebastian.glab@nordicsemi.no>

nordic-segl commented:
I had to add one more commit - remove support for watchdog from Twister configuration yaml for nrf54h20_cpurad.
Otherwise, same tests would run on Application and Radio cores. These on Application core would pass while these on Radio core would fail due to missing overlay files.

nordic-segl commented:
zephyr/samples/drivers/watchdog fails on esp32s3_touch_lcd_1_28 and nrf54l15/cpuflpr
https://github.com/zephyrproject-rtos/zephyr/issues/72655

masz-nordic commented on boards/nordic/nrf54l15pdk/nrf54l15pdk_nrf54l15_cpuflpr.yaml:
```diff
@@ -14,4 +14,3 @@ supported:
   - gpio
   - i2c
   - spi
-  - watchdog
```
If I understand the commit message correctly, the issue is missing `watchdog0` alias. So why not add it instead of disabling support?

nordic-segl replied:
Overlay file for FLPR is also needed in at least two samples/tests.
I don't have experience on using FLPR/PPR to check if watchdog driver works on RISC V.

masz-nordic replied:
Adding this overlay:
https://github.com/zephyrproject-rtos/zephyr/blob/main/samples/drivers/watchdog/boards/nrf54l15pdk_nrf54l15_cpuapp.overlay
but called `nrf54l15pdk_nrf54l15_cpuflpr.overlay` fixes the problem on my side.

nordic-segl replied:
OK. I reverted removal of watchdog from the supported peripherals list for nrf54l15/cpuflpr. I added overlay file as suggested.

nordic-segl committed:
samples: drivers: watchdog: Add overlay files for nrf54l15/cpuflpr

Both nrf54l15pdk/cpuflpr and nrf54l15pdk/cpuflpr_xip have
watchdog on the list of supported periperals.
As a result, Twister tries to build samples/drivers/watchdog
for these two targets.
Since overlays are missing, build is failing.

Add missing overlay files.

Signed-off-by: Sebastian Głąb <sebastian.glab@nordicsemi.no>
